tests/flytekit/unit/core/test_workflows_simulate_remote.py

```diff
@@ -0,0 +1,194 @@
+import io
+import sys
+import typing
+from collections import OrderedDict
+
+import flytekit
+from flytekit.configuration import Image, ImageConfig
+from flytekit.core.condition import conditional
+from flytekit.core.task import task
+from flytekit.core.workflow import workflow
+from flytekit.tools.translator import get_serializable
+
+default_img = Image(name="default", fqn="test", tag="tag")
+serialization_settings = flytekit.configuration.SerializationSettings(
+    project="project",
+    domain="domain",
+    version="version",
+    env=None,
+    image_config=ImageConfig(default_image=default_img, images=[default_img]),
+)
+
```

Comment:
could we add more tests here?
1. dynamic task inside a workflow
2. gate node inside a workflow
3. map task inside a workflow

Reply:
Sure, i would do that.